cleanlab/internal/token_classification_utils.py

```diff
@@ -29,25 +30,27 @@ def get_sentence(words: list) -> str:
 
 
 def filter_sentence(
-    sentences: list, condition: Optional[Callable] = None, return_mask: bool = True
-) -> Union[Tuple[list, list], list]:
+    sentences: List[str],
+    condition: Optional[Callable[[str], bool]] = None,
+    return_mask: bool = True,
+) -> Union[Tuple[List[str], List[bool]], List[str]]:
```

Comment:
For external APIs, do we want to follow the convention that NumPy uses, using `typing.overload` (e.g., https://github.com/numpy/numpy/blob/54c52f13713f3d21795926ca4dbb27e16fada171/numpy/lib/arraysetops.pyi#L121)? I'd be on board with this plan for public APIs: seems like it'll be a common pattern in our library. If we do this, we should document this practice in the development guide, so we can standardize on it.

(See Ryan's message / this Slack thread: https://cleanlabinc.slack.com/archives/C03C42GDTR8/p1661824568503139?thread_ts=1661822614.989389&cid=C03C42GDTR8)

Reply:
Even for internal APIs: on the callee side, (untagged) unions are not a very useful type, and we'll need type assertions (see Ulyana's PR mentioned in that Slack thread); refactoring is an alternative to `typing.overload`.

Reply:
return_mask flag will be removed and the function always return a tuple